packages/web-api/src/methods.ts

```diff
@@ -940,10 +940,16 @@ export interface ViewsPublishArguments extends WebAPICallOptions, TokenOverridab
 }
 
 export interface ViewsUpdateArguments extends WebAPICallOptions, TokenOverridable {
-  view_id: string;
   view: View;
-  external_id?: string;
   hash?: string;
 }
 
+export interface ViewsUpdateArgumentsWithViewId extends ViewsUpdateArguments {
+  view_id: string;
```

Comment:
Another way of fixing #1002 would be to make the `view_id` property optional. 

This change is technically more correct than what we have today. But it does breaks the pattern already established in this file. In fact, there are many arguments interfaces that could benefit from a union of several types like this, but we held ourselves back from making that change because we have an eventual goal of generating these types from an API specification. It's not clear how we'd be able to generate these more complex types from a specification because the specification we have today doesn't express these kinds of interdependencies.

Reply:
if we keep this divided interface, i think it would be more accurate to still include `external_id` as optional when `view_id` exists, and vice versa. One of the two is required, but its okay to specify both.

Reply:
Agreed that it would be more accurate to add `external_id` and `view_id` as optional.

Reply:
I don't think it is a good solution to make both of them optional, as a developer might think both of them can be left blank. Which is completely incorrect!

A better solution than this would be to make `ViewUpdateArugments = ViewsUpdateArgumentsWithViewId | ViewsUpdateArgumentsWithExternalId;` in method.ts so that this wouldn't break the symmetry like suggested here https://github.com/slackapi/node-slack-sdk/pull/1003/files#r417527527

Reply:
@suyashgulati sorry let me clarify the point @aoberoi was trying to make with the optional. 

I think for `ViewsUpdateArgumentsWithViewId` would have `external_id` set to optional, and `ViewsUpdateArgumentsWithExternalId` would set `view_id` to optional. This is because a developer could still include both `external_id` as well as `view_id`

Reply:
Ok I understood now.
Just a query, will adding a view_id makes a difference when I am using external_id for view reference?

Reply:
@aoberoi @stevengill We've reverted this PR but haven't made the field optional yet. Let's re-fix #1002 when you have a chance.